specs/NewWindowSourceFrameInfo.md

```diff
@@ -0,0 +1,113 @@
+            UseShellExecute = true
+        };
+        Process.Start(startInfo);
+        args.Handled = true;
+    }
+};
+```
+```cpp
+// Register a handler for the NewWindowRequested event.
+// This handler will defer the event and check the source frame info to determine
+// how to open the request. Depending on the source frame URI, it will provide
+// a new app window or open using the default browser.
+CHECK_FAILURE(m_webView->add_NewWindowRequested(
+    Callback<ICoreWebView2NewWindowRequestedEventHandler>(
+        [this](ICoreWebView2* sender, ICoreWebView2NewWindowRequestedEventArgs* args)
+        {
+            bool useDefaultBrowser = true;
+
+            Microsoft::WRL::ComPtr<ICoreWebView2NewWindowRequestedEventArgs3> args3;
+            if (SUCCEEDED(args->QueryInterface(IID_PPV_ARGS(&args3)))) {
+                Microsoft::WRL::ComPtr<ICoreWebView2FrameInfo> frame_info;
+                if (SUCCEEDED(args3->get_SourceFrameInfo(&frame_info))
+                    && frame_info)
+                {
+
+                    CHECK_FAILURE(frame_info->get_Source(&source));
+                    useDefaultBrowser = IsThirdPartySource(source.get());
+                }
+            }
+            if (useDefaultBrowser)
+            {
+                // Open the URI in the default browser.
+                ShellExecute(
+                    nullptr, L"open", source.get(), nullptr, nullptr, SW_SHOWNORMAL);
+                CHECK_FAILURE(args->put_Handled(TRUE));
+            }
+            else
+            {
+                wil::com_ptr<ICoreWebView2Deferral> deferral;
+                CHECK_FAILURE(args->GetDeferral(&deferral));
+                AppWindow* newAppWindow = new AppWindow(
+                    m_creationModeId, GetWebViewOption(), L"none", m_userDataFolder, false,
+                    nullptr, true, windowRect, !!shouldHaveToolbar);
+                            newAppWindow->m_isPopupWindow = true;
+                newAppWindow->m_onWebViewFirstInitialized =
+                    [args, deferral, newAppWindow]()
+                {
+                    CHECK_FAILURE(args->put_NewWindow(newAppWindow->m_webView.get()));
+                    CHECK_FAILURE(args->put_Handled(TRUE));
+                    CHECK_FAILURE(deferral->Complete());
+                };
+            }
+            return S_OK;
+        })
+        .Get(),
+    nullptr));
+```
+
+# API Details
+
+```
+/// This is a continuation of the `ICoreWebView2NewWindowRequestedEventArgs` interface.
+[uuid(92f08d94-70bd-4d2b-8332-18bd7d3b2b7c), object, pointer_default(unique)]
+interface ICoreWebView2NewWindowRequestedEventArgs3 :
+    ICoreWebView2NewWindowRequestedEventArgs2 {
+  /// The frame info associated with the frame where the new window request
+  /// originated. See `ICoreWebView2FrameInfo` for details on frame
+  /// properties, such as `Name` and `Source`.
+  [propget] HRESULT SourceFrameInfo([out, retval] ICoreWebView2FrameInfo** frameInfo);
```

Comment:
A FrameInfo is a snapshot in time of info about a frame. It would be good to say explicitly when the snapshot is from. Eg maybe something along the lines of `The frame info of the frame where the new window request originated. The SourceFrameInfo is a snapshot of frame information at the time when the new window was requested.'